model/riscv_insts_fext.sail

```diff
@@ -588,18 +627,23 @@ mapping clause encdec =
 /* Execution semantics ================================ */
 
 function clause execute (F_UN_RM_TYPE_S(rs1, rm, rd, FSQRT_S)) = {
-  let rs1_val_S = nan_unbox (F(rs1));
+  let rs1_val_S = if (sys_enable_zfinx()) then  x_nan_unbox (X(rs1)) else nan_unbox(F(rs1));
   let rm_3b     = encdec_rounding_mode (select_instr_or_fcsr_rm  (rm));
 
   let (fflags, rd_val_S) = riscv_f32Sqrt   (rm_3b, rs1_val_S);
 
   write_fflags(fflags);
-  F(rd) = nan_box (rd_val_S);
+
+  if   ( sys_enable_zfinx() & sizeof(xlen) == 64 )
+  then X(rd) = x_nan_box (rd_val_S)
+  else if ( sys_enable_zfinx() & sizeof(xlen) == 32 )
+  then X(rd) = rd_val_S
+  else F(rd) = nan_box (rd_val_S);
```

Comment:
This paragraph of code is duplicated at least 9 times in this file. I don't know the Sail language that well, but surely there is a way to avoid all this copypasta?